src/game.cpp

```diff
@@ -1681,6 +1681,67 @@ ReturnValue Game::internalTeleport(Thing* thing, const Position& newPos, bool pu
 }
 
 //Implementation of player invoked events
+void Game::playerEquipItem(uint32_t playerId, uint16_t spriteId)
+{
+	Player* player = getPlayerByID(playerId);
+	if (!player) {
+		return;
+	}
+
+	Item* item = nullptr;
+	if (!(item = player->getInventoryItem(CONST_SLOT_BACKPACK))) {
+		return;
+	}
+
+	Container* backpack = nullptr;
+	if (!(backpack = item->getContainer())) {
+		return;
+	}
+
+	static const std::function<Item*(Container*, uint16_t)> searchForItem = [](Container* container, uint16_t itemid) -> Item* {
+		for (ContainerIterator it = container->iterator(); it.hasNext(); it.advance()) {
+			if ((*it)->getID() == itemid) {
+				return *it;
+			}
+		}
+		return nullptr;
+	};
```

Comment:
As this function does not depend on context, you can make it a free function instead. It's better for readability and makes code simpler.

You can use `std::find_if` instead, if this `->iterator()` returns valid C++11 range.